parmed/openmm/parameters.py

```diff
@@ -706,17 +717,18 @@ def _write_omm_residues(self, xml_root, skip_residues, skip_duplicates, valid_pa
                 etree.SubElement(xml_residue, 'Bond', atomName1=bond.atom1.name, atomName2=bond.atom2.name)
             for lonepair in residue.lonepairs:
                 etree.SubElement(xml_residue, 'VirtualSite', self._get_lonepair_parameters(lonepair))
-            for atom in residue.connections:
-                etree.SubElement(xml_residue, 'ExternalBond', atomName=atom.name)
-            if residue.head is not None:
-                etree.SubElement(xml_residue, 'ExternalBond', atomName=residue.head.name)
-            if residue.tail is not None and residue.tail is not residue.head:
-                etree.SubElement(xml_residue, 'ExternalBond', atomName=residue.tail.name)
-            if is_glycam:
+            if not is_glycam:
+                for atom in residue.connections:
+                    etree.SubElement(xml_residue, 'ExternalBond', atomName=atom.name)
+                if residue.head is not None:
+                    etree.SubElement(xml_residue, 'ExternalBond', atomName=residue.head.name)
+                if residue.tail is not None and residue.tail is not residue.head:
+                    etree.SubElement(xml_residue, 'ExternalBond', atomName=residue.tail.name)
+            else:
                 external_bonds = self._get_atoms_with_external_bonds(name)
+                external_bonds = [external_bonds[0], external_bonds[-1]] + external_bonds[1:-1] if len(external_bonds) > 2 and len(external_bonds[-1].name) <= 2 else external_bonds # Hack to get rid of atom ordering diffs
```

Comment:
Can you move this to `_get_atoms_with_external_bonds`?  And what is the `len(external_bonds[-1].name)` checking here?

Reply:
Sorry, that line was accidentally swept into the commit. It's a hack I used to preserve atom ordering, see more [here](https://github.com/ParmEd/ParmEd/pull/1206#issuecomment-963235048).